Tests/GitHubCodespaces.tests.ps1

```diff
@@ -0,0 +1,450 @@
+                }
+            }
+        }
+
+        Context -Name 'When creating a codespace with all possible settings' {
+            BeforeAll {
+                $newGitHubCodespaceParms = @{
+                    # ClientIp = 'TODO ???? - should be instead of rather than in addition to Location, perhaps add some param validation to the function'
+                    # Devcontainer = 'Will add to test in the future when Get-GitHubDevContainer is implemented and the test repo includes one'
+                    DisplayName = 'PowerShellForGitHub pester test'
+                    Location = 'WestUs2'
+                    Machine = 'basicLinux32gb'
+                    NoMultipleRepoPermissions = $true # Not sure how to assert this, but this proves it accepts the switch without error
+                    RetentionPeriod = 10
+                    Timeout = 5
+                    # WorkingDirectory = 'TODO ???? - not sure how to handle this'
+                }
+                $codespace = $repo | New-GitHubCodespace @newGitHubCodespaceParms
+            }
+
+            It 'Should return an object of the correct type' {
+                $codespace | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                # $codespace.devcontainer_path | Should -Be
+                $codespace.display_name | Should -Be $newGitHubCodespaceParms.DisplayName
+                $codespace.idle_timeout_minutes | Should -Be 5
+                $codespace.location | Should -Be $newGitHubCodespaceParms.Location
+                $codespace.machine.name | Should -Be $newGitHubCodespaceParms.Machine
+                $codespace.owner.UserName | Should -Be $script:OwnerName
+                $codespace.repository.name | Should -Be $repo.name
+                $codespace.retention_period_minutes | Should -Be 10
+                $codespace.template | Should -BeNullOrEmpty
+            }
+
+            AfterAll {
+                if ($codespace)
+                {
+                    Remove-GitHubCodespace -CodespaceName $codespace.name -Confirm:$false
+                }
+            }
+        }
+
+        Context -Name 'When creating a codespace with default settings with Repository Elements' {
+            BeforeAll {
+                $newGitHubCodespaceParms = @{
+                    RepositoryName = $repo.name
+                    OwnerName = $script:organizationName
+                }
+                $codespace = New-GitHubCodespace @newGitHubCodespaceParms
+            }
+
+            It 'Should return an object of the correct type' {
+                $codespace | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                $codespace.display_name | Should -Not -BeNullOrEmpty
+                $codespace.repository.name | Should -Be $repo.name
+                $codespace.idle_timeout_minutes | Should -Be 30
```

Comment:
Same comment here about not testing this value since the default isn't documented.  Here and elsewhere in the file where I may have missed adding the comment as well.